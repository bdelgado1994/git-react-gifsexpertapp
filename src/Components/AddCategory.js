import React, { useState } from "react";
import PropTypes from 'prop-types';

export default function AddCategory({ setCategories, categories }) {
    const [inputValue, setInputValue] = useState('');
    const handleChage = (e) => {
        setInputValue(e.target.value);
        console.log('Handle Change Cambiado');
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue !== undefined) {
            setCategories([
                inputValue, ...categories
            ])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                type={'text'}
                value={inputValue}
                onChange={handleChage}
            />
        </form>
    );
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};