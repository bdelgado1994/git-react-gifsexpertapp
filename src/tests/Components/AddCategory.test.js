import React from 'react';
import { shallow } from 'enzyme'
import AddCategory from '../../Components/AddCategory'
describe('Prueba Unitaria de AddCategory', () => {
    const setCategory = () => { };
    const wrapper = shallow(<AddCategory setCategories={setCategory} />);
    test('Prueba del componente', () => {

        expect(wrapper).toMatchSnapshot();
    })

})
