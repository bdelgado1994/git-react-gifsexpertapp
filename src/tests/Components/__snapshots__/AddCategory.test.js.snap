// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Unitaria de AddCategory Prueba del componente 1`] = `
<form
  onSubmit={[Function]}
>
  <img
    src="https://pokeapi.co/api/v2/pokemon/132"
  />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
